instant-xml/src/impls.rs

```diff
@@ -281,45 +281,6 @@ impl<'xml> FromXml<'xml> for String {
     const KIND: Kind = Kind::Scalar;
 }
 
-impl<'xml> FromXml<'xml> for &'xml str {
-    #[inline]
-    fn matches(id: Id<'_>, field: Option<Id<'_>>) -> bool {
-        match field {
-            Some(field) => id == field,
-            None => false,
-        }
-    }
```

Comment:
Yes, this is a general fix -- it was kind of related because while trying to deal with decoding I was running into some issues with lifetimes but those issues are no longer relevant. I still feel (like you) that the impl makes it too easy write something simple that works most of the time but then suddenly fails, and just failing at compile-time by requiring `Cow` feels like a better trade-off.